crates/api/tests/traps.rs

```diff
@@ -69,8 +69,10 @@ fn test_trap_trace() -> Result<(), String> {
     assert_eq!(trace.len(), 2);
     assert_eq!(trace[0].module_name().unwrap(), "hello_mod");
     assert_eq!(trace[0].func_index(), 1);
+    assert_eq!(trace[0].func_name(), "hello");
     assert_eq!(trace[1].module_name().unwrap(), "hello_mod");
     assert_eq!(trace[1].func_index(), 0);
+    assert_eq!(trace[1].func_name(), "");
```

Comment:
Could we default to the export name in this case?

Reply:
It is coming from name section, it is hard to setup relationship between export name and function.

Reply:
If we need to store a lookup map going from function index -> export names to implement this then I think it would be worth doing.  Showing an "empty" entry point just because a user didn't give the exported function a local name will *really* confuse users.

Reply:
For context, many native code debuggers will display export names in absence of symbolic information for a function.

Reply:
What shall take precedence: name section or export name. If two exports defined pointing to same name, which name shall appear in the trace when function is called for particular export?

Reply:
Those are great questions.

I personally think name section takes precedence and the export name is used as a last resort.  If there's some symbolic name we can use, we should use it.  Function indexes should only be displayed to users when there's no way to attach a symbolic name to the frame (the function is not exported and there's no name section).

With multiple exported names, I think a stable heuristic is all that's needed (i.e. we only insert into the map once, thus a stable ordering based on first encountered in the table).